shared/src/store/action-reducers/utils/calculate-treatments.spec.ts

```diff
@@ -23,38 +25,49 @@ interface Catering {
     patientIds: UUID[];
 }
 
+/**
+ * Asserts that adding the specified {@link caterings} to the {@link beforeState} results in the {@link newState}.
+ * If the {@link beforeState} has already caterings in it, these will not be removed.
+ */
 function assertCatering(
     beforeState: ExerciseState,
     newState: ExerciseState,
     caterings: Catering[]
 ) {
-    const shouldState = produce(newState, (draftState) => {
-        caterings.forEach((catering) => {
-            const expectedAssignedPatients: Mutable<UUIDSet> = {};
-            catering.patientIds.forEach((patientId) => {
-                expectedAssignedPatients[patientId] = true;
-            });
-            expect(
-                newState[catering.catererType][catering.catererId]!
-                    .assignedPatientIds
-            ).toStrictEqual(expectedAssignedPatients);
-            draftState[catering.catererType][
+    // Add all caterings to the before state and look whether the result is the newState
+    const shouldState = produce(beforeState, (draftState) => {
+        for (const catering of caterings) {
+            // Update all the patients
+            const patients = catering.patientIds.map((patientId) =>
+                getElement(draftState, 'patients', patientId)
+            );
+            for (const patient of patients) {
+                patient[
+                    catering.catererType === 'materials'
+                        ? 'assignedMaterialIds'
+                        : 'assignedPersonnelIds'
+                ][catering.catererId] = true;
+                if (catering.catererType === 'personnel') {
+                    patient.isBeingTreated = true;
+                }
+            }
+            // Update the catering element
+            const cateringElement = getElement(
+                draftState,
+                catering.catererType,
                 catering.catererId
-            ]!.assignedPatientIds = {};
-        });
-        const patientIds = caterings.flatMap((catering) => catering.patientIds);
-        patientIds.forEach((patientId) => {
-            expect(newState.patients[patientId]!.isBeingTreated).toStrictEqual(
-                true
             );
-            draftState.patients[patientId]!.isBeingTreated = false;
-        });
+            for (const patientId of catering.patientIds) {
+                cateringElement.assignedPatientIds[patientId] = true;
+            }
+        }
         return draftState;
     });
-    expect(shouldState).toStrictEqual(beforeState);
+    expect(newState).toStrictEqual(shouldState);
```

Comment:
`expect(received).toStrictEqual(expected)` Is this still the correct order here?

Reply:
So it's correct now, right? I've always had trouble thinking about this specific order in this test.

Reply:
I think it is correct now (I believe it was the other way around before and confused me a lot ^^)

Reply:
I think you're correct. The test was much more confusing before this PR as it removed part of the state and checked that one.